Unity/SymbolResolver.cs

```diff
@@ -12,6 +12,9 @@ public T ResolveFunction<T>(string name)
         {
             var address = Resolve(name);
 
+            if (address == IntPtr.Zero)

```

Comment:
`SymbolResolver` is not intended to throw when a symbol is not found, as this is not considered an exceptional situation (since it can be used for checking whether or not a symbol exists). Checking for `null` and throwing should be done at the call site instead.

Alternatively, a new `TryResolve*` group of methods can be added, with this new "throw on failure" behaviour being used for the `Resolve*` variants.

Reply:
I found throwing an exception made it easier to track down errors when running on versions that aren't supported. If a new version of unity comes out and renames a symbol, its useful for it to throw with an exception that logs the symbol name rather then having a null exception later.

Reply:
Probably best if we go the `TryResolve` route for non-throwing situations then. Something like:
```CSharp
public unsafe T* Resolve<T>(string name)
    where T : unmanaged
{
    if (TryResolve<T>(name, out T* address))
        return address;

    throw new UnresolvedSymbolException(name);
}

public unsafe bool TryResolve<T>(string name, out T* address)
    where T : unmanaged
{
    address = (T*)Resolve(name);
    return address != null;
}
```